dask/bag/core.py

```diff
@@ -817,16 +818,31 @@ def topk(self, k, key=None, split_every=None):
         return self.reduction(func, compose(func, toolz.concat), out_type=Bag,
                               split_every=split_every, name='topk')
 
-    def distinct(self):
+    def distinct(self, key=toolz.identity):
         """ Distinct elements of collection
 
         Unordered without repeats.
 
+        Parameters
+        ----------
+        key: callable
+            Defines uniqueness such as in `toolz.unique`
```

Comment:
I suspect that there are dask.bag users that are not familiar with `toolz`.  It would probably be good to fully specify what this parameter without reference to toolz